pkg/sensors/exec/cgroups_test.go

```diff
@@ -70,6 +117,125 @@ func cgroupRmdir(t *testing.T, cgroupfsPath string, hierarchy string, dir string
 	return err
 }
 
+func mountCgroup(t *testing.T, root string, kind string) (string, error) {
+	err := os.MkdirAll(root, os.ModeDir)
+	if err != nil {
+		return "", err
+	}
+
+	err = syscall.Mount(root, root, kind, 0, "")
+	if err != nil {
+		return "", err
+	}
+
+	t.Logf("Mount %s  type=%s succeeded", root, kind)
+	return root, nil
+}
+
+func umountCgroup(t *testing.T, root string) error {
+	err := syscall.Unmount(root, 0)
+	if err != nil {
+		t.Logf("Failed to unmount %s: %v", root, err)
+	} else {
+		t.Logf("Unmount %s succeeded", root)
+	}
+	return err
+}
+
+func getCgroupEventOpAndPath(t *testing.T, msg *grpcexec.MsgCgroupEventUnix, cgroupHierarchy []cgroupHierarchy) (ops.CgroupOpCode, string) {
+	if msg.Common.Op != ops.MSG_OP_CGROUP {
+		return ops.CgroupOpCode(ops.MSG_OP_CGROUP_UNDEF), ""
+	}
+
+	cgrpPath := cgroups.CgroupNameFromCStr(msg.Path[:processapi.CGROUP_PATH_LENGTH])
+	require.NotEmpty(t, cgrpPath)
+
+	cgrpName := cgroups.CgroupNameFromCStr(msg.CgrpData.Name[:processapi.CGROUP_NAME_LENGTH])
+	require.NotEmpty(t, cgrpName)
+	require.Equal(t, cgrpName, filepath.Base(cgrpPath))
+
+	op := ops.CgroupOpCode(msg.CgrpOp)
+	st := ops.CgroupState(msg.CgrpData.State).String()
+	logger.GetLogger().WithFields(logrus.Fields{
+		"cgroup.event":     op.String(),
+		"PID":              msg.PID,
+		"NSPID":            msg.NSPID,
+		"cgroup.IDTracker": msg.CgrpidTracker,
+		"cgroup.ID":        msg.Cgrpid,
+		"cgroup.state":     st,
+		"cgroup.level":     msg.CgrpData.Level,
+		"cgroup.path":      cgrpPath,
+	}).Info("Received Cgroup event")
+
+	// match only our target cgroup paths
+	if strings.HasPrefix(cgrpPath, filepath.Join("/", cgroupHierarchy[0].path)) == false {
+		return ops.CgroupOpCode(ops.MSG_OP_CGROUP_UNDEF), ""
+	}
+
+	require.NotZero(t, msg.PID)
+	require.NotZero(t, msg.Cgrpid)
+	require.NotZero(t, msg.CgrpidTracker)
+	require.NotEqualValues(t, msg.CgrpData.State, ops.CGROUP_UNTRACKED)
+	require.NotZero(t, msg.CgrpData.Level)
+
+	return op, cgrpPath
+}
+
+func requireCgroupEventOpMkdir(t *testing.T, msg *grpcexec.MsgCgroupEventUnix, cgrpMapPath string) {
+	require.EqualValues(t, ops.CGROUP_NEW, msg.CgrpData.State)
+
+	looked, err := cgrouptrackmap.LookupTrackingCgroup(cgrpMapPath, msg.CgrpidTracker)
+	assert.NoError(t, err)
+	if looked == nil {
+		t.Fatalf("Failed to find tracking cgroupID=%d in bpf-map=%s", msg.CgrpidTracker, cgrpMapPath)
+	}
+	require.EqualValues(t, msg.CgrpData.Level, looked.Level)
+	require.EqualValues(t, msg.CgrpData.Name, looked.Name)
```

Comment:
Why do you mix `require` and `assert` here? I understand that require simply fails the test. Can a failure in `cgrouptrackmap.LookupTrackingCgroup` be acceptable for this test?